ingestion_tools/scripts/common/finders.py

```diff
@@ -218,3 +234,22 @@ def load(
         if source.get("literal"):
             return BaseLiteralValueFinder(**source["literal"])
         raise Exception("Invalid source type")
+
+
+class MultiSourceFileGlobFinder(DefaultImporterFactory):
```

Comment:
Nice! My only real comment is that this config is a little bit fragile - it lets us list multiple globs that can find different kinds of files to import for a single annotation. For example we can have config like this:
```
alignments:
- metadata:
    ....
  sources:
    source_multi_glob:
       list_globs:
         - "{run_name}/tilts/*.xf"
         - "{run_name}/tilts/tilt.com"
```
, but it doesn't let us specify anything EXTRA each glob. For example we can't specify a regex filter or format information for each glob. This means that we're 100% stuck relying on filename extensions to understand how to read each file returned by this finder. I think that most tomogram reconstruction workflows are pretty standardized and already heavily reliant on file names being named in special ways, but wanted to get a comment from @uermel on whether this is reasonable.

Reply:
I can't guarantee that this will be sufficient in every case, pretty much every expectation I had concerning users' data organization has been proven wrong by an edge case. 

That said, I think within a group of runs globs will suffice. Will there be the ability to define include/exclude patterns for groups of runs? In that case one could define multiple alignments blocks for different groups of runs, where the glob patterns can change. I think that should be sufficient.

Reply:
> Will there be the ability to define include/exclude patterns for groups of runs for the alignment blocks? In that case one could define multiple alignments blocks for different groups of runs, where the glob patterns can change. I think that should be sufficient. In the worst case we may have to use csv-mapped values to include in these glob strings.

Yes, we should be able to use parent filters like we do in other places. But, list globs don't support match_regex at the moment.

Reply:
If we don't have any outliers for this at the moment, as we comfortable with this solution for now, and maybe extending it in the future when the need arises? @jgadling

Reply:
Ok, now that `AlignmentImporter` has a custom `ImporterFactory` I have two concerns about where it's at right now:
1. It takes exactly the same `sources` configs as our default `ImporterFactory` but it has different *behavior* - I think this is confusing -- for tiltseries for example, every file found by every glob in the `source_multi_glob` finder will become a new Tiltseries, but for Alignments, the files found by every glob in the `source_multi_glob` config will all be *combined* into a single Alignment. I'd prefer if the interface for the alignment configs somehow let users know that it works *differently* for alignments than it does for other imported types.
2. If (when) we find outliers in the future, my concern is that we'll have to change the format of the `sources` blocks in all existing ingestion configs to match the format supported by our newer/better/outlier-handling finder. It's not a lot of work to write a finder with a more flexible config structure right now, but it's a lot more work migrating all this later.
Maybe something along the lines of:
```yaml
  sources:
  - fan_in_finder:
    - glob: "{run_name}/tilts/*.xf"
      include_regex: ".*FILTERED.*"
      format: "xf"
    - glob: "{run_name}/tilts/tilt.com"
      format: "tilt.com"
```
**Caveat though** -- Now that `Alignments` has a custom finder, we have the ability to add **additional** source types without breaking reverse compatibility, so item 2 isn't a hard blocker, but I'm curious to know what you think.

Reply:
As the new importer is handling the usecases for current and foreseeable future, I think this might be good enough to get us to the finish line. Especially as it only has internal users who would be interacting with it at the moment who are aware of the alignment ingestion behaviour.  

I was also originally envisioning something with a key that defines the  explicitly for the finder. But, that doesn't provide a lot of added benefit right now, so I would lean towards working on this enhancement in the next phase. 

I believe there shouldn't be a big enough difference in the amount of effort needed to migrate the work later to newer pattern between doing it now or later, as it is being done in two steps, instead of one single step.

Reply:
I'd prefer if we handled these cases now, but if you want to proceed with this as it is, I think we need to *document* the differences between the multi-glob finder behavior in Alignments vs other importers. Maybe a big code block in `alignments.py` or maybe a markdown file. The "behavior is weird but it's just us" thing falls apart as we onboard more/different people to the project, so we need to leave some notes behind.